libs/libc/wchar/lib_wcwidth.c

```diff
@@ -0,0 +1,253 @@
+/****************************************************************************
+ * libs/libc/wchar/lib_wcwidth.c
+ *
+ * Copyright © 2005-2014 Rich Felker, et al.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining
+ * a copy of this software and associated documentation files (the
+ * "Software"), to deal in the Software without restriction, including
+ * without limitation the rights to use, copy, modify, merge, publish,
+ * distribute, sublicense, and/or sell copies of the Software, and to
+ * permit persons to whom the Software is furnished to do so, subject to
+ * the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+ * CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <wchar.h>
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+static const unsigned char g_table[] =
+{
```

Comment:
how do you plan to maintain these tables?

Reply:
should I place the content g_table[] to a separate header file ?

Reply:
The code base comes from musl:
https://github.com/bminor/musl/blob/master/src/ctype/wide.h
https://github.com/bminor/musl/blob/master/src/ctype/nonspacing.h
https://github.com/bminor/musl/blob/master/src/ctype/wcwidth.c#L3-L9
so, it may better to ask musl owner to improve the design and comment, so we can incorporate the improvement back.

Reply:
it's our problem to reformat imported code and make maintenance harder.
it isn't a problem in musl.

Reply:
the original code also lack any information about these two tables. Why you consider is the reformat problem?

Reply:
at least it's easier to copy changes from the upstream if we don't reformat.

Reply:
Yes, but the code incorporated into nuttx has to conform the coding style which we must follow.